frontend/src/pages/search/index.jsx

```diff
@@ -1,25 +1,25 @@
 import React from 'react'
 import Link from 'next/link'
 import { object, string } from 'prop-types'
-import useSWR, { SWRConfig } from 'swr'
+import { SWRConfig } from 'swr'
+import useSWRInfinite from 'swr/infinite'
 
 import Page from '@components/Page'
 import { useSearchQuery } from '@contexts/SearchContext'
-import ProjectCard from '@components/ProjectCard'
-import { meiliIndex } from '@utils/meili'
+import CardsGrid, {
+  cardsPerRow,
+  getKey,
+  gridFetcher,
+  useUpdateBeforeReachingLimit,
+} from '@components/CardsGrid'
 
 import styles from './index.module.scss'
 
-const fetchSearch = async query => {
-  const searchResult = await meiliIndex.search(query)
-  return searchResult.hits
-}
-
 export const getServerSideProps = async ({ query }) => {
   // '*' or '' means return everything but '' can't be used as a SWR cache key
   const { q = '*' } = query
 
-  const hits = await fetchSearch(q)
+  const hits = await gridFetcher({ query: q })
```

Comment:
This makes for some confusing reading. 

![image](https://user-images.githubusercontent.com/206854/203849380-6eb4c7a2-b409-49d7-ad02-e5249cdd2d72.png)